packages/atclient/tests/test_atserver_message.c

```diff
@@ -47,8 +47,10 @@ int main() {
 
 uint8_t test_1a_long_prompt() {
   const char *TAG = BASE_TAG("1a");
-  char *buffer = "@foobar@data:baz";
-  const uint16_t len = strlen(buffer);
+
+  const uint16_t len = 17;
+  char buffer[len];
+  memcpy(buffer, "@foobar@data:baz\n", len);
 
   struct atserver_message message = atserver_message_parse(buffer, len);
```

Comment:
Made all of these buffers in the tests non-constant now that atserver_message_parse modifies them.